types/broadcast.go

```diff
@@ -0,0 +1,69 @@
+package types
+
+import (
+	"crypto/ed25519"
+	"fmt"
+)
+
+type WakeupBroadcast struct {
+	Sequence Varu64
+	Root
+	Signature [ed25519.SignatureSize]byte
+}
+
+func (w *WakeupBroadcast) ProtectedPayload() ([]byte, error) {
+	buffer := make([]byte, w.Sequence.Length()+w.Root.Length())
```

Comment:
I don't remember why we don't recycle buffers on these protected payloads, but it might be a nice thing to do in a future PR to keep allocations down.